File: GXpert/GXpert.Web/Modules/Attendance/PollResponse/PollResponseColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Attendance.Columns;

[ColumnsScript("Attendance.PollResponse")]
[BasedOnRow(typeof(PollResponseRow), CheckNames = true)]
public class PollResponseColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string PollQuestion { get; set; }
    public string StudentPrn { get; set; }
    [EditLink]
    public string Response { get; set; }
    public int ResponseTimeInSeconds { get; set; }
    public int Score { get; set; }
    public int LiveSessionLogId { get; set; }
    public string ActivationDeviceId { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public bool IsActive { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/PollResponseColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PollResponseRow } from "./PollResponseRow";

export interface PollResponseColumns {
    Id: Column<PollResponseRow>;
    PollQuestion: Column<PollResponseRow>;
    StudentPrn: Column<PollResponseRow>;
    Response: Column<PollResponseRow>;
    ResponseTimeInSeconds: Column<PollResponseRow>;
    Score: Column<PollResponseRow>;
    LiveSessionLogId: Column<PollResponseRow>;
    ActivationDeviceId: Column<PollResponseRow>;
    InsertDate: Column<PollResponseRow>;
    InsertUserId: Column<PollResponseRow>;
    UpdateDate: Column<PollResponseRow>;
    UpdateUserId: Column<PollResponseRow>;
    IsActive: Column<PollResponseRow>;
}

export class PollResponseColumns extends ColumnsBase<PollResponseRow> {
    static readonly columnsKey = 'Attendance.PollResponse';
    static readonly Fields = fieldsProxy<PollResponseColumns>();
}